### DISCUSSION TITLE
SOLVED! Possible Bug? when using Identify -format and the compression attribute?

### DISCUSSION THREAD

**Discussion by JoshuaInDC** [ImageMagick/ImageMagick#1898] [General] [answered]
Hi Everyone,

I can't figure out why Imagemagick is giving two different outputs for the same file. See my commands and outputs, and tell me if I have missed something, or if there is a bug.

identify -format %p-%C\n CompressionError.tif
      and
identify -format %p-%[compression]\n CompressionError.tif

I thought that I would receive the same output from both commands, but I don't. Are these two parameters different? Does one apply to the input file, and another to the output file? I don't notice any differentiation on the [manual website](https://imagemagick.org/script/escape.php). Any suggestions?

See the batch output below, and note that I am using the old way of calling identify.

C:\Temp\Compression Improvement Test>identify -format %p-%C\n CompressionError.tif
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1037.
0-LZW
1-LZW
2-LZW
3-LZW
4-LZW

C:\Temp\Compression Improvement Test>identify -format %p-%[compression]\n CompressionError.tif
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1037.
0-LZW
1-None
2-None
3-None
4-None

Thanks!

**Comment by dlemstra** [comment]
Could you also share your `CompressionError.tif` file?

**Reply by JoshuaInDC** [reply]
Here is the link. This was a file that was processed and cleaned up in Photoshop I believe.
https://1drv.ms/u/s!Ak3IHZNUKeG2id8gcmT32x_2XVxczg?e=GZCWwt

**Comment by urban-warrior** [comment, marked as answer]
By default, image identification is light weight and fast.  It only guarantees accurate results for a limited number of image attributes such as width, height, scene, and format.  See https://imagemagick.org/script/identify.php.  To retrieve accurate image properties, add +ping to your command-line.  Its a slower process, but guarantees accurate results:

```
$ identify +ping -format "%p-%[compression]\n" CompressionError.tif 
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1037.
0-LZW
1-None
2-None
3-None
4-None

$ identify +ping -format "%p-%C\n" CompressionError.tif
identify: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/1037.
0-LZW
1-None
2-None
3-None
4-None
```

**Reply by JoshuaInDC** [reply]
+PING is the solution!
Thanks for that! 
[Learn more here by searching +ping.](http://www.imagemagick.org/Usage/basics/) 
and also
[Learn about +ping](https://imagemagick.org/script/command-line-options.php#ping)
